src/views/ops/Template/components/AddVariableDialog.vue

```diff
@@ -62,6 +62,11 @@ export default {
     editVariable(form) {
       const i = this.variables.findIndex(item => item.var_name === this.variable.var_name)
       this.variables.splice(i, 1, form)
+      const count = this.variables.filter(value => value.var_name === form.var_name || value.name === form.name).length
+      // 不允许有相同的变量名
+      if (count > 1) {
+        this.variables.splice(i, 1)
+      }
       this.iVisible = false
     }
   }
```

Comment:
There are some issues with the current code that need to be addressed.

1. Incorrect use of `indexOf` method for finding variables when checking if it's already exists:

    ```javascript
      const i = this.variables.findIndex(item => item.var_name === variable.var_name)
    ```

Correcting it would make it look like:
```javascript
const i = this.variables.findIndex((item) => item.var_name === variable.var_name);
```
This corrects the spelling mistake and improves the function name so it fits proper context in English syntax.

2. There seems there is no reference to an object named `variable`.

3. The loop should start from 0 rather than index `-1`, since we do not know which entry to replace initially (since "i" starts at zero based on iteration).

Please review these points for any modifications you might recommend.